### PR TITLE
LPS-194407 Internal server error when using an ERC with more than 75 characters

### PR DESCRIPTION
**Pull request by Artartur**
Related Issue: [LPS-194407](https://liferay.atlassian.net/browse/LPS-194407)

### CODE REVIEW

**Comment by Artartur** [comment]
ci:test:sf

**Comment by Artartur** [comment]
ci:test:object

**Comment by Artartur** [comment]
ci:test:relevant

File: modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectDefinitionExternalReferenceCodeException.java [hunk 1]
```diff
@@ -22,24 +22,46 @@ public ForbiddenUnmodifiableSystemObjectDefinitionExternalReferenceCode(
 			String externalReferenceCode) {
 
 			super(
+				StringBundler.concat(
+					"Forbidden unmodifiable system object definition external ",
+					"reference code ", externalReferenceCode),
 				StringBundler.concat(
 					"Forbidden unmodifiable system object definition external ",
 					"reference code ", externalReferenceCode));
```
Comment (murilostodolni) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectDefinitionExternalReferenceCodeException.java:30]: We need to create the language key, something like this:
`forbidden-unmodifiable-system-object-definition-external-reference-code=Forbidden unmodifiable system object definition external reference code.`

And follow the same pattern the comment above, please

File: modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectDefinitionExternalReferenceCodeException.java [hunk 2]
```diff
@@ -22,24 +22,46 @@ public ForbiddenUnmodifiableSystemObjectDefinitionExternalReferenceCode(
...
+	}
+
 	public static class MustNotStartWithPrefix
 		extends ObjectDefinitionExternalReferenceCodeException {
 
 		public MustNotStartWithPrefix() {
-			super("The prefix L_ is reserved for Liferay");
+			super(
+				"The prefix L_ is reserved for Liferay",
+				"The prefix L_ is reserved for Liferay");
```
Comment (murilostodolni) [inline modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectDefinitionExternalReferenceCodeException.java:52]: We need to create the language key, something like this:
`the-prefix-x-is-reserved-for-liferay=The prefix {0} is reserved for Liferay`

And follow the same pattern the comment above, please

Comment (murilostodolni) [reply modules/apps/object/object-api/src/main/java/com/liferay/object/exception/ObjectDefinitionExternalReferenceCodeException.java:52]: One more thing: I observed that we use "L_" in several places in the code. We can add a new constant in `ObjectDefinitionConstants.java`, something like this:
`public static final String PREFIX_SYSTEM_OBJECT_DEFINITION_EXTERNAL_REFERENCE_CODE = "L_"`

and replace in all place that use the "L_" hard code.

**Review by murilostodolni — CHANGES_REQUESTED** [review]
Hey @Artartur

I finished my review. Feel free to talk with me if necessary. Thank you

**Comment by murilostodolni** [comment]
Hey @Artartur 

I resent the PR to: https://github.com/liferay-objects/liferay-portal/pull/2514.

Feel free to contact me if necessary.

**Comment by murilostodolni** [comment]
ci:close

**Pull request closed without merging**